types/react-dom/index.d.ts

```diff
@@ -53,49 +53,51 @@ export function unstable_renderSubtreeIntoContainer<P>(
     container: Element,
     callback?: (component?: Component<P, ComponentState> | Element) => any): Component<P, ComponentState> | Element | void;
 
+export type Container = Element | Document | DocumentFragment;
```

Comment:
React is missing `DocumentFragment` internally but it is allowed: 
- https://github.com/facebook/react/blob/v16.9.0/packages/react-dom/src/client/ReactDOM.js#L452-L457
- https://github.com/facebook/react/blob/v17.0.2/packages/react-dom/src/client/ReactDOMRoot.js#L203-L207